Add whylabs and panther labs case studies (#587)
pulumi/pulumi-hugo: themes/default/content/case-studies/whylabs.md

As more developers joined WhyLabs and the company began to think about multi-cloud environments, they needed a modernized approach that would enable the company to scale its cloud infrastructure with the speed and agility required of an innovative startup. WhyLabs and its “two-pizza teams” needed a platform that would enable them to apply existing software engineering best practices to their infrastructure, and the platform needed to support any cloud.

## Bringing Pulumi Onboard

Andy Dang is a Lead Engineer and a co-founder of WhyLabs. He first heard about Pulumi from a colleague at AI2. He says, “I was blown away by the multi-language experience as well as the various innovations, such as its best-in-class support for Kubernetes. Those qualities, as well as support for multiple cloud providers made me decide to go with Pulumi.”

## Adopting Cloud Engineering Best Practices with Pulumi

> Moving to Pulumi allowed WhyLabs to adopt cloud engineering best practices out-of-the-box. Cloud engineering teams apply standard software engineering practices and tools uniformly across infrastructure management, application development, and security to tame the complexity of delivering and managing modern cloud applications. Key advantages for WhyLabs were that they could:

Author: ```suggestion
Moving to Pulumi allowed WhyLabs to adopt cloud engineering best practices out-of-the-box. [Cloud engineering]({{< relref "/cloud-engineering" >}}) teams apply standard software engineering practices and tools uniformly across infrastructure management, application development, and security to tame the complexity of delivering and managing modern cloud applications. Key advantages for WhyLabs were that they could:
```

### Review

Pulumi allows the infrastructure code review process to fit naturally within the GitLab code review process. This is particularly helpful when sensitive changes, such as those that are security related, need approval. Because infrastructure is treated like any other piece of code Pulumi and GitLab make it easy to route proposed changes to the correct people for review. Not only does this make the review process itself easier, but it means that changes are well documented and easily audited.

### Test and Deploy

Since infrastructure is software with Pulumi, it can be tested like software through a CI/CD pipeline. WhyLabs uses Pulumi and GitLab to run automated unit tests on infrastructure and Policy as Code tests that check for common human errors such as leaving an S3 bucket open to the public. The pipeline also runs integration tests that can detect issues with the stack early in the development process by provisioning ephemeral environments. Furthermore, WhyLabs developers can use Pulumi to deploy private environments to test their logic, enabling them to iterate on complex cloud environments without causing issues to the production environment.

### Security

> At WhyLabs, developers have their own stacks where they can test out deployments first, before they’re merged with a production stack. It’s important to be able to track those deployments and to see who triggered each deployment. WhyLabs uses Pulumi to embed security controls in the pipeline to keep track of these events. The upshot is that stacks are secured automatically, rather than relying on manual, after-the-fact fixes. In addition, WhyLabs uses Pulumi to manage secrets with AWS Secrets Manager and to manage VPC security and network control.

Reviewer: ```suggestion
At WhyLabs, developers have their own stacks where they can test out deployments before they’re merged with a production stack. WhyLabs uses Pulumi to embed security controls in the pipeline to keep track of events. For example, they can track all deployments and see who triggered each deployment. Stacks are secured automatically rather than relying on manual, after-the-fact fixes. In addition, WhyLabs uses Pulumi to manage secrets with AWS Secrets Manager and to manage VPC security and network control.
```
